### DISCUSSION TITLE
Installation instruction's function code snippet should be different for installed with go & mention add env variable for editor permanently in bashrc

### DISCUSSION THREAD

**Discussion by BenjaminWang236** [antonmedv/llama#72] [General]
I think the installation instructions should include:
1. If installed with go, and llama is at `~/go/bin/llama`, then the function code snippet added to `.bashrc` should be `cd "$(~/go/bin/llama "$@")"` instead of `cd "$(llama "$@")"`. At least that's how it was for me.
3. Permanently adding editor environmental variable to `.bashrc` (works with micro) like "You can permanently add it by adding the `export LLAMA_EDITOR=micro` line to your `.bashrc`", etc

My environment is WSL - ubuntu, if that makes a difference

PS: I like llama, pretty intuitive to use

**Comment by antonmedv** [comment]
This is just linux 101. Don’t think README should teach *nix basics.

**Reply by Slackadays** [reply]
I don't know about this. If a user falls into a trap by reading only the README, then that's the README's problem because we shouldn't make assumptions about the user.

**Reply by antonmedv** [reply]
> shouldn't make assumptions about the user

Without assumptions about the user it is not possible to do anything.

**Reply by Slackadays** [reply]
Although this is true, I just think it's unreasonable to assume the user knows about these little details because many don't. We might have the perspective of power users, but most users aren't power users.

**Reply by antonmedv** [reply]
So what exactly you think will be nice to add? I also want to keep readme clean and minimalistic.

**Reply by Slackadays** [reply]
Here's two potential ideas:

Add the golang case as a comment in the code block, like
```bash
function ll {
  cd "$(llama "$@")"
  #cd "$(~/go/bin/llama "$@")" if you installed with go
}
```
and change `Put the next function into the .bashrc or a similar config:` to `Put the next function into .bashrc or a similar config (and export LLAMA_EDITOR=your_favorite_editor if you wish):`.

Or, you could move the entire `.bashrc` section into a Wiki page all about installation where you have lots of space for details.

**Reply by antonmedv** [reply]
>   #cd "$(~/go/bin/llama "$@")" if you installed with go


Maybe better explain what is $PATH var and how *nix systems lookup binaries with some docs on how go install work?

**Reply by Slackadays** [reply]
That would work, so maybe like:

Use `cd "$(~/go/bin/llama "$@")"` instead if you installed with go, as by default, the system doesn't search for binaries there.

**Reply by antonmedv** [reply]
I don’t like it.
